vars/branchRepo.groovy

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/groovy
+
+def call(String gitRepoUrl, String gitRef, String checkoutDir, String branchName, Closure body) {
+    checkoutGitRepo {
+        repoUrl = gitRepoUrl
+        branch = gitRef
+        targetDir = checkoutDir
+    }
+    dir(checkoutDir) {
+        sshagent(['jenkinsgithub']) {
```

Comment:
Yes, was already thinking about that. I don't however want to add all these to the signature so might add a config/parameters object like we have done in others e.g https://github.com/feedhenry/fh-pipeline-library/blob/master/vars/fhBuildNode.groovy#L3.